go/private/fake_bazel_features.bzl

```diff
@@ -0,0 +1,35 @@
+# bazel_features when used from a WORKSPACE rather than bzlmod context requires a two-step set-up (loading bazel_features, then calling a function from inside it).
+# rules_go only has one-step set-up, and it would be a breaking change to require a second step.
+# Accordingly, we supply a fake implementation of bazel_features which is only used when using rules_go from a WORKSPACE file,
+# to avoid complicating code in rules_go itself.
+# We just implement the checks we've seen we actually need, and hope to delete this completely when we are in a pure-bzlmod world.
+
+_FAKE_BAZEL_FEATURES = """bazel_features = struct(
+  cc = struct(
+    find_cpp_toolchain_has_mandatory_param = {find_cpp_toolchain_has_mandatory_param},
+  )
+)
+"""
+
+def _fake_bazel_features_impl(rctx):
```

Comment:
This looks pretty real to me. Maybe call it a `polyfill` instead? Thanks for going the extra mile!